sch_fq.c

```diff
@@ -556,13 +785,39 @@ static struct sk_buff *fq_dequeue(struct Qdisc *sch)
 			return NULL;
 		}
 	}
+    }
+
 	f = head->first;
 
+	int rValue = valuePresentinArray(f->socket_hash, pFlowid);
+	
+	printk("rValue is   : %d \n ", rValue);
+	
+	//printk("barrier value  : %d \n ", barrier[dcounter]);
+
+/* Breach and membership of the flow is checked once it is satisfied all the flows are added to co-flow set at once
+	if( (rValue != -1) && (barrier[dcounter] == 3) )
+	{
+	printk("Breach Occured \n");
+	barrier[dcounter] = 0;
+	head->first = f->next;
```

Comment:
Do you want to do this linked list manipulation here or handle it in your promotecoflows function?

Reply:
I'd prefer the latter, it's more uniform and easier to understand.